libraries/lib-snapping/SnapUtils.h

```diff
@@ -87,12 +87,23 @@ struct SNAPPING_API SnapRegistryItem : public Registry::SingleItem
    virtual SnapResult SingleStep(const AudacityProject& project, double time, bool upwards) const = 0;
 };
 
-class ConstantMultiplierSnapItem final : public SnapRegistryItem
+using MultiplierFunctor = std::function<double(const AudacityProject&)>;
+
+class TimeInvariantSnapItem final : public SnapRegistryItem
```

Comment:
I would much prefer the way it is done in previous commit. This still exposes details outside the translation unit (and in case we will want to add `SnapRegistryItem`s in different libraries - we will have to expose it outside the library too). However version with two classes express intention clearly and avoids (very small) storage and branching overhead.

I do not have any strong opinion on that though, so it is up to you.